src/lib/erc20.ts

```diff
@@ -102,20 +261,170 @@ const multiCoinContracts = {
   // aurora: '0xc9bA77C9b27481B6789840A7C3128D4f691f8296',
 } satisfies { [key in Chain]: `0x${string}` }
 
-const networkToken = {
-  arbitrum: ADDRESS_ZERO,
-  avalanche: ADDRESS_ZERO,
-  bsc: ADDRESS_ZERO,
-  celo: '0x471ece3750da237f93b8e339c536989b8978a438',
-  ethereum: ADDRESS_ZERO,
-  fantom: ADDRESS_ZERO,
-  harmony: ADDRESS_ZERO,
-  polygon: ADDRESS_ZERO,
-  moonbeam: ADDRESS_ZERO,
-  optimism: ADDRESS_ZERO,
-  gnosis: ADDRESS_ZERO,
-} satisfies { [key in Chain]: `0x${string}` }
+export async function balanceOf({
+  client,
+  address: walletAddress,
+  token,
+}: {
+  client: PublicClient
+  chain: Chain
+  address: Address
+  token: Token
+}): Promise<Balance | boolean> {
+  try {
+    const result = await readContract(client, {
```

Comment:
`viem` automatically does JSON RPC batch when calling this in a `Promise.all`. Will check if wrapping `readContract` inside another function then calling `Promise.all` prevents batching. If not I will change it to use `call` function for uniformity.

Reply:
Would a `batch` arg on `readContract` help if you want to opt-out of JSON-RPC batching?

Reply:
@jxom that would be helpful!
Currently in all 3 cases it will automatically batch the requests into one if I have `batch: true` in my `publicClient` config?

```ts
const balances = await Promise.all(
  [/** */].map(token =>
      readContract(client, {
        abi: ERC20_ABI,
        address: getAddress(token.address),
        functionName: "balanceOf",
        args: [getAddress(address)],
      })
    ),
  ),
);
```

vs `readContract` wrapped inside `balanceOf` helper:

```ts
const balances = await Promise.all(
  [/** */].map(token => balanceOf({ client, chain, address, token })),
);
```
or
```ts
const balances = await Promise.allSettled(
  [/** */].map(token => balanceOf({ client, chain, address, token })),
);
```

This is how I was going to test if it does batch or not. Only way I know how to test it is in the browser console network tab.